openide.awt/src/org/netbeans/modules/openide/awt/DefaultAWTBridge.java

```diff
@@ -68,11 +72,12 @@ public JMenuItem createMenuPresenter (Action action) {
         return item;
     }
     
+    @Override
     public Component createToolbarPresenter(Action action) {
         AbstractButton btn;
-        if (action instanceof BooleanStateAction) {
+        if ((action instanceof BooleanStateAction) || (action.getValue(Actions.ACTION_VALUE_TOGGLE) != null)) {
```

Comment:
Nice idea, but `getValue()` is final in `SystemAction` as well as `getProperty()`. I can put `ACTION_VALUE_TOGGLE` into values as a marker, but still the `DefaultAWTBridge` has to obtain the value. I cannot synchronize `setBooleanState` and `getBooleanState` with `Action.SELECTED_KEY` value I think because there are overrides of `getBooleanState` which directly return the result (i.e. `ProjectTab,NodeSelectionProjectAction`.

I fear that it needs to reference the class in the implementation. But using bytecode manipulation, I could split the deprecated APIs away into a 'compat' module, so at least the API would be free of `BooleanStateAction` references.